src/core/textures/BaseTexture.js

```diff
@@ -579,8 +587,15 @@ export default class BaseTexture extends EventEmitter
     {
         if (this.imageUrl)
         {
-            delete BaseTextureCache[this.imageUrl];
-            delete TextureCache[this.imageUrl];
+            // remove this texture from the global texture cache
+            if (BaseTextureCache[this.imageUrl] === this)
+            {
+                delete BaseTextureCache[this.imageUrl];
+            }
+            if (TextureCache[this.imageUrl] && TextureCache[this.imageUrl].baseTexture === this)
```

Comment:
Because `fromLoader()` adds both BaseTexture and Texture to their respective caches under both `name` and `imageUrl` keys. I was hoping to avoid adding a bunch of new variables just to have each track both keys for later removal.